lib/src/data/models/tmdb_api_model.dart

```diff
@@ -0,0 +1,21 @@
+import 'movie_model.dart';
+
+class TMDBApiModel {
+  late int page;
+  late int totalResults;
+  late int totalPages;
+  List<MovieModel> results = <MovieModel>[];
```

Comment:
Why you are defining this as a late? You can directly create a constructor like
```
TMDBApiModel({
     required this.page,
     required this.totalResults,
     required this.totalPages,
     required this.result
})
```

Reply:
Review the name also, this is more like a ResponseModel, you can avoid to specify the name, remember that if in the future we change TMDB to StarWars Api, you should change the name of the class, make sense?

Reply:
Okay, i've change the name but if i do de parameters required what i gonna do is that the API Repository take responsability of null checking. It's that okay?

Reply:
You don't need to validate that, because you know that when you call the API and is a success, you always going to get a model with page, totalResults, totalPages, and result

Reply:
Okay, understood. But just in case what happens if the API changes the way of send the results? (like you say with the change of the name earlier) Shouldn't i check somehow if the response come like i think is gonna be?

Reply:
Other thing, in the factory i returned a list of movies (i don't know if that is even okay). So what i do is first generate the object `responseModel` and later i call a method of the class that return me a list of movies to send it in the DataState. It's that okay? Because i don't see a lot of sense to create the class `ResponseModel` if i'm gonna send only a list of movies